tools/2025-03-workflow-changes.md

```diff
@@ -0,0 +1,260 @@
+
+### Design options considered but not chosen
+
+* Adopt a new automation tool like [Just](https://github.com/casey/just) or [Task](https://taskfile.dev/), and deprecate Make. These tools have advantages, but a side-effect of adopting a new tool would likely be the existence of a new tool in our toolbox without the removal of Make, which would increase complexity. Make currently meets our needs for invoking developer automation.
+* Remove everything from GitHub workflows except the invocation of another tool: GitHub workflows have advantages when setting up the automation environment, e.g. retrieving credentials, cloning repos, configuring runners, installing tools, etc.
+
+## Security
+
+### Least Privilege
+
+Workflows will continue to adhere to the least privilege feature in which they declare which permissions are required by the workflow.
+
+### Identity
+
+Where possible, the automation logic will use the GitHub CLI, which has identity built in. Developers would log into the CLI when running code on their machines, which would allow them to run the logic on their forks. When running as part of at GitHub workflow, the GitHub CLI context is already set up and no further configuration is required.
+
+## Development plan
+
+### Docs: Contributing Automation Code
+
+A new folder exists in the `radius/docs` folder that provides a `README.md` file with guidance on creating and editing GitHub workflows and other repo automation like `make`.
+
+### PR Template
+
+Modify the PR template checklist for each repository that contains workflows to include items similar to the following:
+
+Workflow modifications have been tested on a repo fork and the changes conform to the workflow development guidance (add hyperlink to the guidance )  
+
+* [ ] This PR contains no modifications to GitHub workflows or repo automation.  
+* [ ] Changes to GitHub workflows, actions, and other repo automation have been tested on a repo fork.
+
+### Existing GitHub Workflows
+
+Existing workflows will evolve to implement the design principles. For example, if a new step is being added to an existing workflow, the step would be added via a Make command, which adheres to the principle that all logic is runnable on a developer machine.
+
+### New GitHub Workflows
+
+New GitHub workflows will adhere to the design principles.
+
+## Open Questions and Actions
+
+* Which actions cannot be performed by the GitHub CLI? How will identity be provided to them?
+* Provide guidance on which git/GitHub CLI commands allow a PR reviewer to check out a PR to their own fork.
+* We need a prototype of moderate complexity that implements these principles.
+* Can we separate the scheduling from the workflows so that forks do not automatically run the workflows?
+
+## Design Review Notes
+
+* Add an example of moving logic into a Make command and show the before and after.
```

Comment:
Where will this example be added?

Reply:
I added an example, but how about adding a real-world example into the documentation? The workflow updates planned in the current sprint will almost certainly provide several good examples.